### ISSUE TITLE
Can This Library Simplify File Downloading?

### ISSUE THREAD

**Issue by machineghost** [jprichardson/node-fs-extra#842]
- **Operating System:** - Not relevant (question/documentation issue)
- **Node.js version:** - Not relevant (question/documentation issue)
- **`fs-extra` version:**  - Not relevant (question/documentation issue)

The Stack Overflow question https://stackoverflow.com/questions/37614649/how-can-i-download-and-save-a-file-using-the-fetch-api-node-js provides the following recipe for fetching a file using the (boring/non-promisified) `fs` module:

```
const downloadFile = (async (url, path) => {
  const res = await fetch(url);
  const fileStream = fs.createWriteStream(path);
  await new Promise((resolve, reject) => {
      res.body.pipe(fileStream);
      res.body.on("error", reject);
      fileStream.on("finish", resolve);
    });
});
```

My question (as someone who's not an expert on file buffers/streams/etc.) is, can the `createWriteStream` method of this library (or any other method) be used to avoid the need to create a new Promise in the above code.

In other words, if I use `node-fs-extra` instead of `fs`, can the above code be simplified further?  If so, it might make a nice addition to the documentation, since this is probably a fairly common scenario.

**Comment by RyanZim** [comment]
No, streams are event-based, which does not have a 1-to-1 mapping to promises. `fs-extra` does not provide anything to make this cleaner.

**Comment by machineghost** [comment]
That makes sense, was just hoping there was something I was missing.  Thanks for the answer!

**Comment by jimmywarting** [comment]
fyi, writeFile & appendFile both support async iterable and typedarrays so it's possible to just write web streams

```js
const res = await fetch(url)
await fs.writeFile(res.body)
console.log('done')
```

i think writeFile is supperior to createWriteStream as it supports both node and web streams
